jina/clients/base/websocket.py

```diff
@@ -54,16 +54,35 @@ async def _get_results(
                     WebsocketClientlet(url=url, logger=self.logger)
                 )
 
-                request_buffer: Dict[str, asyncio.Future] = dict()
+                request_buffer: Dict[
+                    str, asyncio.Future
+                ] = dict()  # maps request_ids to futures (tasks)
+                request_buffer[
```

Comment:
what is the logic here?

Reply:
what is this empty request id?

Reply:
Usually, a failure response has the same id as the request that caused it.
When there is a network failure during endpoint discovery, we don't have access to the request (and its id), so I put `''`.

With some effort it could probably be done that also in this case the response contains the request id, but i think logically it makes sense that it doesn't have this id, because the error is not really connected to the request. Any request would have caused the error, the Flow was broken from the beginning.
I could also put in a random id instead of empty, but then the client would disregard this response because it doesn't know the id, so that would necessitate further changes on the client.

That's why there appears this empty id in the client, it is the id that a failure response caused during endpoint discovery will have.

Not too pretty overall, I agree. The websocket client seems to need a lot of hacks...